hibernate-core/src/main/java/org/hibernate/tool/schema/internal/SchemaDropperImpl.java

```diff
@@ -155,12 +167,36 @@ public void doDrop(Metadata metadata, boolean dropSchemas, Dialect dialect, Targ
 			);
 		}
 
-		for ( Namespace namespace : database.getNamespaces() ) {
-			if ( dropSchemas ) {
-				if ( namespace.getName().getSchema() == null ) {
-					continue;
+		if ( tryToDropCatalogs || tryToDropSchemas ) {
+			Set<Identifier> exportedCatalogs = new HashSet<Identifier>();
+
+			for ( Namespace namespace : database.getNamespaces() ) {
+				if ( tryToDropCatalogs ) {
```

Comment:
I believe these should be inverted from create.  Meaning, I believe you should drop the schemas first and then drop the catalogs.